common/shared.c

```diff
@@ -363,7 +363,7 @@ int ft_init_av(void)
 		ret = fi_av_insert(av, fi->dest_addr, 1, &remote_fi_addr, 0, NULL);
 		if (ret != 1) {
 			FT_PRINTERR("fi_av_insert", ret);
-			return ret;
+			return -1;
```

Comment:
When fi_av_insert returns 0, we could print a proper error message. I think FI_PRINTERR would just print "0(success)" which is not that useful. Also when the return value is not 0 I think we should pass along the error code instead of returning -1.